feature-libs/checkout/base/components/guards/checkout-steps-set.guard.ts

```diff
@@ -104,6 +106,11 @@ export class CheckoutStepsSetGuard implements OnDestroy {
   }
 
   protected isStepSet(step: CheckoutStep): Observable<GuardResult> {
+    if (!this.hasDeliveryItemsInCart) {
```

Comment:
Can we move the handling of 'pickup in store' to a guard defined within the OPF checkout sub-library, instead of modifying a guard used by all checkout flows? I believe changing the shared guard could introduce a potential breaking change. Since we only want to handle 'pickup in store' when the OPF flow is active, isolating it in the OPF-specific logic might be a better approach.